aspnetcore/tutorials/first-mongo-app/first-mongo-app.md

```diff
@@ -0,0 +1,165 @@
+
+Open the Powershell or command promt and run the following command
+
+```console
+mongod --dbpath <Enter the path>
+```
+
+(Replace dbpath value with your own path) This will connect to MongoDB on port 27017.
+
+Open another Powershell instance and run 'mongo'.	
+
+```console
+mongo
+```
+
+This will connect to default test database. Run the following command on > (Powershell)
+
+```console
+useBookstoreDb
+```
+
+This will create a database named 'BookstoreDb' if it doesn't already exists, else it will open the existing Database for transactions.
+
+We can create a collection using following command.
+
+```console
+db.createCollection('Books')
+```
+The Schema for the Books collection can be defined using following command from the powershell.
+
+```console
+db.Books.insert({'BookId':1,'BookName':'Design Patterns','Price':3000,'Category':'Computers', 'Author':' Ralph Johnson'})
+```
+
+To see the entries in your database, you can use following command
+
+```console
+db.Books.find({})
+```
+
+The following result will be displayed.
+
+```console
+db.Books.find({})
+{ "_id" : ObjectId("5ad08f91a4f1c236ef1b4fff"), "BookId" : 1, "BookName" : "Design Patterns", "Price" : 3000, "Category" : "Computers", "Author" : " Ralph Johnson" }
+```
+
+The schema will add _id property. This property will be an ObjectId which will be generated automatically.
+
+Now that our database is ready, we can start creating our Web API Application using .Net Core.
+
+### Creating the .Net Core Web API Application
+
+You need Visual Studio with .Net core SDK for creating the .Net Application. For the purpose of this article, I will be using Visual Studio 2017.
+
+In your visual studio, go to File > New > Project and select "ASP.NET Core Web Application from .NET Core tab.
+
+![New Project](_static/newProject.png)
+
+Name the Application "BookMongo" (or whatever you like) and click ok. This will open a windows for selecting templates. Choose the "API" template and Click ok.
+
+![Select Template](_static/apiTemplate.png)
+
+This will create your Web API project.
+
+No click right on your Project name in the solution explorer and go to Manage Nuget Packages.
+
+Search for mongocsharpdriver and install the package.
+
+![Package Manager](_static/packageManager.png)
+
+### Adding Data Model
+
+Now, in the project, add a Models folder. In this folder, add a new class file of name Book.cs. Add the following code in this file.
+
+[!code-csharp[](~/tutorials/first-mongo-app/sample/BookstoreAPI/Models/Book.cs?name=snippet_1)]
+
+The class also contains **Id** property of the type **ObjectId**. This property is mandatory so that the CLR object can be mapped with Collection in MongoDB. The class contains properties having the **BsonElement** attribute applied on it. This represent the mapped property with the MongoDB collection.
+
+### Writing a class for CRUD operations
+
+Now, add a DataAccess.cs class file in the Models folder with the following code in it
```

Comment:
- File and folder names, such as DataAccess.cs and Models, should be italicized.
- End the sentence with a colon.